beep/structure.py

```diff
@@ -132,29 +132,37 @@ def from_file(cls, path, validate=False):
         else:
             raise ValueError("{} does not match any known file pattern".format(path))
 
-    def get_interpolated_discharge_cycles(self, v_range=None, resolution=1000):
+    def interpolated_cycles(self, v_range, resolution, step_type='discharge'):
         """
-        Gets interpolated cycles just for the discharging states.
+        Gets interpolated cycles for the step specified, charge or discharge.
 
         Args:
             v_range ([Float, Float]): list of two floats that define
                 the voltage interpolation range endpoints.
             resolution (int): resolution of interpolated data.
+            step_type (str): which step to interpolate i.e. 'charge' or 'discharge'
 
         Returns:
             pandas.DataFrame: DataFrame corresponding to interpolated values.
         """
+        if step_type is 'discharge':
+            group = self.data.groupby(["cycle_index", "step_index"]).filter(
+                determine_whether_step_is_discharging).groupby("cycle_index")
+        elif step_type is 'charge':
+            group = self.data.groupby(["cycle_index", "step_index"]).filter(
+                determine_whether_step_is_charging).groupby("cycle_index")
+        else:
+            raise NotImplementedError
```

Comment:
Can we make the error message more explicit?  Also might want to make it a ValueError, unless we're planning on implementing another step_type.

Reply:
Agreed